src/random_ray/random_ray_simulation.cpp

```diff
@@ -111,47 +111,83 @@ void validate_random_ray_inputs()
     }
   }
 
-  // Validate solver mode
-  ///////////////////////////////////////////////////////////////////
-  if (settings::run_mode == RunMode::FIXED_SOURCE) {
-    fatal_error(
-      "Invalid run mode. Fixed source not yet supported in random ray mode.");
-  }
-
   // Validate ray source
   ///////////////////////////////////////////////////////////////////
 
   // Check for independent source
   IndependentSource* is =
     dynamic_cast<IndependentSource*>(RandomRay::ray_source_.get());
   if (!is) {
-    fatal_error(
-      "Invalid ray source definition. Ray source must be IndependentSource.");
+    fatal_error("Invalid ray source definition. Ray source must provided and "
+                "be of type IndependentSource.");
   }
 
   // Check for box source
   SpatialDistribution* space_dist = is->space();
   SpatialBox* sb = dynamic_cast<SpatialBox*>(space_dist);
   if (!sb) {
     fatal_error(
-      "Invalid source definition -- only box sources are allowed in random "
-      "ray "
-      "mode. If no source is specified, OpenMC default is an isotropic point "
-      "source at the origin, which is invalid in random ray mode.");
+      "Invalid ray source definition -- only box sources are allowed.");
   }
 
   // Check that box source is not restricted to fissionable areas
   if (sb->only_fissionable()) {
-    fatal_error("Invalid source definition -- fissionable spatial distribution "
-                "not allowed for random ray source.");
+    fatal_error(
+      "Invalid ray source definition -- fissionable spatial distribution "
+      "not allowed.");
   }
 
   // Check for isotropic source
   UnitSphereDistribution* angle_dist = is->angle();
   Isotropic* id = dynamic_cast<Isotropic*>(angle_dist);
   if (!id) {
-    fatal_error("Invalid source definition -- only isotropic sources are "
-                "allowed for random ray source.");
+    fatal_error("Invalid ray source definition -- only isotropic sources are "
+                "allowed.");
+  }
+
+  // Validate fixed sources
+  ///////////////////////////////////////////////////////////////////
+  if (settings::run_mode == RunMode::FIXED_SOURCE) {
+    if (model::external_sources.size() < 1) {
+      fatal_error("Must provide a particle source (in addition to ray source) "
+                  "in fixed source random ray mode.");
+    }
+
+    for (int i = 0; i < model::external_sources.size(); i++) {
+      Source* s = model::external_sources[i].get();
+
+      // Check for independent source
+      IndependentSource* is = dynamic_cast<IndependentSource*>(s);
+
+      if (!is) {
+        fatal_error("Only IndependentSource fixed source types are allowed in "
+                    "random ray mode");
+      }
+
+      // Check for isotropic source
+      UnitSphereDistribution* angle_dist = is->angle();
+      Isotropic* id = dynamic_cast<Isotropic*>(angle_dist);
+      if (!id) {
```

Comment:
Note that I changed this line (and others) in accordance with [rule ES.87](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es87-dont-add-redundant--or--to-conditions) from the C++ core guidelines.